components/script/dom/bindings/reflector.rs

```diff
@@ -13,7 +13,7 @@ use std::default::Default;
 
 /// Create the reflector for a new DOM object and yield ownership to the
 /// reflector.
-pub fn reflect_dom_object<T, U>(
+pub fn reflect_dom_object<#[must_root] T, #[must_root] U>(
```

Comment:
I think the right solution for `Box<T>` where T is actually a rooted type is creating a `#[allow_unrooted_interior] struct Unrooted<#[must_root] T>(Box<T>)` and using that instead of `Box<T>`.

Reply:
Sounds reasonable. If I remember correctly, this is the only usage of `Box<T>` with `#[must_root] T`, though. At least, the only usage reported by the compiler when I was gradually fixing the violations. So I can be wrong.